Meter.c

```diff
@@ -124,29 +125,51 @@ static void BarMeterMode_draw(Meter* this, int x, int y, int w) {
 
    // First draw in the bar[] buffer...
    int offset = 0;
+   double barLen=0;
    for (uint8_t i = 0; i < this->curItems; i++) {
       double value = this->values[i];
       if (isPositive(value) && this->total > 0.0) {
          value = MINIMUM(value, this->total);
-         blockSizes[i] = ceil((value / this->total) * w);
+         barLen = ((value / this->total) * w);
+         blockSizes[i] = ceil(barLen);
       } else {
          blockSizes[i] = 0;
       }
       int nextOffset = offset + blockSizes[i];
       // (Control against invalid values)
       nextOffset = CLAMP(nextOffset, 0, w);
+
       const Settings* settings = this->host->settings;
+      const char* bars[7][8] = {
+         {"|","|","|","|","|","|","|","|"},
+         {"#","#","#","#","#","#","#","#"},
+         {"⡀","⡄","⡆","⡇","⣇","⣧","⣷","⣿"},
+         {"░","░","▒","▒","▓","▓","█","█"},
+         {"▏","▎","▍","▌","▋","▊","▉","█"},
+         {"▁","▂","▃","▄","▅","▆","▇","█"},
+         {"▌","▌","▌","▌","█","█","█","█"}
+      };
```

Comment:
Include the whitespace character for the various sets, as for e.g. braile you want to use the "zero dots" codepoint, instead of the normal blank character for consistency.

Also you can use `" ||||||||"`, which while using one byte more is more compact to write. Furthermore, your type should be:

```suggestion
      static const char* bars[7] = {
         " ||||||||",
         " ########",
         " ⡀⡄⡆⡇⣇⣧⣷⣿",
         " ░░▒▒▓▓██",
         " ▏▎▍▌▋▊▉█",
         " ▁▂▃▄▅▆▇█",
         " ▌▌▌▌████",
      };
```
(Used blank for all of them for now, please update the blanks for the graphical ones)

Reply:
Will commit everything bundled in one commit 

i'll change the data type from instead of char to wchar so that we won't have to worry about converting char to wchar on everytime requested

Reply:
thanks for the changes i was quite uncertain while writing this :D

also i had to ask why don't we use normal space ?
and which is the utf8 character that you're referring to instead of the space.

Reply:
For Braille you can use this one:
https://www.compart.com/en/unicode/U+2800

Reply:
Reason for using other ones is for consistency, because sometimes the spacehas a different width in fonts compared to the default blank …

Reply:
> Reason for using other ones is for consistency, because sometimes the spacehas a different width in fonts compared to the default blank …

damn i didn't even think about this you're a genius :exploding_head:

Reply:
i will check the website for other characters don't worry about it :D
btw i will have to keep spaces for | and # right?

Reply:
Yes. Only for the grapical sets you may want to change things if necessary.

Reply:
@BenBE 

> Reason for using other ones is for consistency, because sometimes the spacehas a different width in fonts compared to the default blank …

This is not an issue in monospace terminals. The real issue you should worry, however, is that certain block characters will occupy two character width in East Asian locales. It's true for these blocks `▏▎▍▌▋▊▉█`.

This is known as [East Asian Width](https://www.unicode.org/reports/tr11/) property. And you use `wcwidth(3)` to get this property in C.

Data from Unicode:
```text
2580..258F     ; A  # So    [16] UPPER HALF BLOCK..LEFT ONE EIGHTH BLOCK
2590..2591     ; N  # So     [2] RIGHT HALF BLOCK..LIGHT SHADE
2592..2595     ; A  # So     [4] MEDIUM SHADE..RIGHT ONE EIGHTH BLOCK
2596..259F     ; N  # So    [10] QUADRANT LOWER LEFT..QUADRANT UPPER RIGHT AND LOWER LEFT AND LOWER RIGHT
```

Reply:
hey there tried finding other characters but couldn't find others
i thought it would be easy 

what do i do for the remaining characters??
people online suggested to use spaces

Reply:
> hey there tried finding other characters but couldn't find others i thought it would be easy
> 
> what do i do for the remaining characters?? people online suggested to use spaces

For block characters, Unicode recommendeds U+00A0 No-break Space for empty block.

Reply:
For completeness here is another set: The upper filled blocks. Requires Unicode 13 or later.

`\u00A0\u2594\U0001FB82\U0001FB83\u2580\U0001FB84\U0001FB85\U0001FB86\u2588;`
`▔🮂🮃▀🮄🮅🮆█`

Reply:
oh hey there thanks exlorer will add the no break space for all other places

and will add the other character set you recommended

i had to ask should i add it in reverse order??

Reply:
or would the bars be starting from upside down?

Reply:
Just made the latest commit see how it feels would love to know suggestions
@BenBE @Explorer09 
Thanks for helping me out

Reply:
I might be crazy suggesting this, but with Unicode 16 another set could become available.
This is like the braille set `⡀⡄⡆⡇⣇⣧⣷⣿` but with dots replaced with blocks.

`\u00A0\U0001CEA3\u2596\U0001CD48\u258C\U0001CDC0\u2599\U0001CDE4\u2588`
`𜺣▖𜵈▌𜷀▙𜷤█`

This is pretty new, so not many fonts support rendering these. [Cascadia Code](https://github.com/microsoft/cascadia-code) is one of the fonts supporting these new characters.

Reply:
@BenBE what do you think about this ??

should i add it ?
if yes,
should i replace it instead of the currently used characters for braille?

or should we add utf16 later ??

Reply:
btw adding utf16 is as easy as just adding new line in the array

Reply:
I think, Unicode 13 (March 2020) is fine. For Unicode 16 (September 2024) we can add this as a commented out inclusion, but should not compile it in by default (people shall edit their code if they wish to use it). Regarding the Unicode 16 variant I also have my doubts, if this set should be abbreviated or if there is a slightly extended version. Also for braille you could even do 0 to 8 dots starting at the bottom and adding the bottom-left most empty dot.

P.S.: There's a tool called `charmap` that can be useful to look up these characters.

Reply:
so is this done??

Reply:
if there is anything yet to add please do let me know